oqsprov/oqs_prov.h

```diff
@@ -25,11 +25,20 @@
 #define OQS_PROVIDER_BASE_VERSION_STR OQSPROVIDER_VERSION_TEXT
 
 #ifdef NOPUBKEY_IN_PRIVKEY
-#define OQS_PROVIDER_VERSION_STR OQS_PROVIDER_BASE_VERSION_STR "-nopub"
+#define NOPUBKEY_IN_PRIVKEY_STR "-nopub"
 #else
-#define OQS_PROVIDER_VERSION_STR OQS_PROVIDER_BASE_VERSION_STR
+#define NOPUBKEY_IN_PRIVKEY_STR ""
 #endif
 
+#ifdef USE_ENCODING_LIB
+#define ENCODING_LIB_STR "-encoding_support"
+#else
+#define ENCODING_LIB_STR ""
+#endif
+
+#define OQS_PROVIDER_VERSION_STR OQS_PROVIDER_BASE_VERSION_STR NOPUBKEY_IN_PRIVKEY_STR ENCODING_LIB_STR
```

Comment:
OK - I asked for it -- but isn't this a bit "too much"? Presence of encoding support is already shown in line 469, right? Preferably we can drop this (as well as "-nopub" (would be nice if we could achieve agreement on this soon -- not duplicating the public key in the private key could be default as far as I'm concerned: What's your take, @dstebila @bhess? If I'm not mistaken, this behaviour is no default only because we wanted interoperability between oqs-openssl111 and oqs-provider -- but we may slowly want to drop that, right?)

Reply:
Removed the redundant string now (also -nopub).

Re. public key in private key: I imagine it can be useful to be able to get the public key directly from a PKCS#8 structure, e.g. for a signer who wants to test sign/verify without regenerating the pk. An advantage of the IETF drafts implemented in this PR is that the private keys are structured to have an OPTIONAL public key. If one choses to include the OPTIONAL part, there won't be a duplication of the public key. If one just concatenates the raw sk/pk, there will be duplication in some cases like Kyber/SPHINCS+.